helix-view/src/document.rs

```diff
@@ -1093,6 +1098,19 @@ impl Document {
         }
     }
 
+    pub fn version_control_head(&self) -> Option<String> {
+        self.version_control_head
+            .clone()
+            .map(|head| (*head).load().as_ref().as_ref().to_owned())
```

Comment:
I'm afraid we can't do much. But this looks ugly.

Reply:
It was quite ugly, I must agree. But we just gave it a facelift! :lipstick: 

```rust
pub fn version_control_head(&self) -> Option<Arc<Box<str>>> {
    self.version_control_head.as_ref().map(|a| a.load_full())
}
```

Thanks for the feedback! :smile: